### PR TITLE
Add `apm_config.obfuscation.cache.max_size` to set the maximum size of obfuscation query cache

### PR DESCRIPTION
**Pull request by keisku**
### What does this PR do?

Add `DD_APM_OBFUSCATION_CACHE_MAX_SIZE` (= `apm_config.obfuscation.cache.max_size`) to set the maximum size of obfuscation cache.

### Motivation

The cache size was fixed to 5000000 (= 5000 bytes * 1000 queries) because [the maximum length of `resource` length is 5000](https://github.com/DataDog/documentation/blob/3e6ebe979fb3cac7001d3a432b2fdd9c965f02b4/content/en/tracing/troubleshooting/_index.md?plain=1#L221) and 1000 queries were expected to be enough for caching. However, DBM was not considered when determining this value.

#31336 enabled obfuscation cache by default for the DBM context other than APM. I think 5000000 bytes is not ideal value for all DBM customers. Therefore, I'd rather make the cache size be configurable.

### Describe how you validated your changes

We can see this debug log with the following Dockerfile and docker-compose.yaml.

```bash
docker run --name datadog-agent -d --cgroupns host --pid host \
-v /var/run/docker.sock:/var/run/docker.sock:ro \
-v /proc/:/host/proc/:ro -v /sys/fs/cgroup/:/host/sys/fs/cgroup:ro \
-e DD_API_KEY=xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx \
-e DD_APM_OBFUSCATION_CACHE_MAX_SIZE=654321 \
datadog/agent-dev:keisku-obfuscation-cache-size-py3

docker exec -it datadog-agent agent config
.....
    cache:
      enabled: true
      max_size: 654321
```

### Possible Drawbacks / Trade-offs

The default size is unchanged.

### CODE REVIEW

File: comp/trace/config/testdata/full.yaml [hunk 1]
```diff
@@ -91,3 +91,4 @@ apm_config:
       luhn: true
     cache:
       enabled: true
+      max_size: 5555555
```
Comment (louis-cqrl) [inline comp/trace/config/testdata/full.yaml:94]: why this value is not equal to `5000000` ?

**Review by louis-cqrl — APPROVED** [review]
LGTM overall

File: pkg/config/config_template.yaml [hunk 2]
```diff
@@ -1265,6 +1265,10 @@ api_key:
   ##        Enables caching obfuscated statements. Currently supported for SQL and MongoDB queries.
   ##        Enabled by default.
   #         enabled: true
+  ##        @param DD_APM_OBFUSCATION_CACHE_MAX_SIZE - integer - optional - default: 5000000
+  ##        The maximum size of the cache in bytes. The maximum allowed resource length is 5000.
+  ##        We can store a minimum of 1000 queries (= 5000000 / 5000) by default.
```
Comment (buraizu) [inline pkg/config/config_template.yaml:1270]:
```suggestion
  ##        Datadog stores a minimum of 1000 queries (5000000 / 5000) by default.
```

**Review by buraizu — CHANGES_REQUESTED** [review]

File: comp/trace/config/testdata/full.yaml (hunk 1 above)
Comment (keisku) [reply comp/trace/config/testdata/full.yaml:94]: Looks env var is prioritized. I added a test case that expects `o.Cache.MaxSize` equals to `5555555`.

**Review by ajgajg1134 — APPROVED** [review]

**Review by vickenty — APPROVED** [review]

**Review by buraizu — APPROVED** [review]

**Pull request merged**